internal/core/metadata/rest_addressable.go

```diff
@@ -190,105 +173,95 @@ func restGetAddressableByName(w http.ResponseWriter, r *http.Request) {
 	vars := mux.Vars(r)
 	dn, err := url.QueryUnescape(vars[NAME])
 	if err != nil {
-		LoggingClient.Error(err.Error())
-		http.Error(w, err.Error(), http.StatusServiceUnavailable)
+		httpErrorHandler.HandleOne(w, err, nil, errorconcept.Default.ServiceUnavailable)
 		return
 	}
 
 	op := addressable.NewNameExecutor(dbClient, dn)
 	result, err := op.Execute()
 	if err != nil {
-		LoggingClient.Error(err.Error())
-		if err == db.ErrNotFound {
-			http.Error(w, err.Error(), http.StatusNotFound)
-		} else {
-			http.Error(w, err.Error(), http.StatusServiceUnavailable)
-		}
+		httpErrorHandler.HandleOne(
```

Comment:
This will fit on a single line.  Calling out this single instance -- there may be others that you can condense.